crates/observe/src/distributed_tracing/trace_id_format.rs

```diff
@@ -129,11 +129,19 @@ where
         event: &Event<'_>,
     ) -> fmt::Result {
         let trace_id = Span::current().context().span().span_context().trace_id();
+        let mut line = String::new();
+
+        // now let the normal formatter do all the fancy stuff and dump it into a String
+        let format_res = self.inner.format_event(ctx, Writer::new(&mut line), event);
         if trace_id != TraceId::INVALID {
-            write!(writer, "[trace_id={trace_id}] ")?;
+            // remove the new line the default formatter added
+            if line.ends_with('\n') {
+                line.pop();
+            }
+            // append trace id and a newline
+            line.push_str(&format!(" [trace_id {trace_id}]\n"));
```

Comment:
This way it would look like the regular log format, no? Might be nicer to not cause any confusion.
```suggestion
            line.push_str(&format!(" trace_id={trace_id}\n"));
```

Reply:
Double clicking in opensearch with that format works just fine for me. (e.g. in this [doc](https://aws-es.cow.fi/_dashboards/app/discover#/doc/5ed837d0-4a67-11ef-8b11-a98c7c46873d/cowlogs-staging-2025.07.09?id=lwRk75cBEHdYNVOVn84r)).
If double clicking is only an issue when selecting in a terminal I'd prefer to optimize for the experience of using the logs in opensearch.